plugins/importexport/onix30/filter/MonographONIX30XmlFilter.inc.php

```diff
@@ -517,7 +518,7 @@ function createProductNode($doc, $submission, $publicationFormat) {
 		$resourceVersionNode = $doc->createElementNS($deployment->getNamespace(), 'ResourceVersion');
 		$supportingResourceNode->appendChild($resourceVersionNode);
 		$resourceVersionNode->appendChild($this->_buildTextNode($doc, 'ResourceForm', '01')); // Linkable resource
-		$resourceVersionNode->appendChild($this->_buildTextNode($doc, 'ResourceLink', $publication->getLocalizedCoverImageUrl($context->getId(), $publication->getData('locale'))));
+		$resourceVersionNode->appendChild($this->_buildTextNode($doc, 'ResourceLink', $publication->getLocalizedCoverImageUrl($context->getId(), $pubLocale)));
```

Comment:
also here, maybe to use getData instead of getLocalizedCoverImageUrl

Reply:
In this case `getLocalizedCoverImageUrl` is doing some extra steps to generate a URL to the cover image - so we either need to do that here in the plugin with `getData`, or create something like `getCoverImageUrl` that uses `getData` instead of `getLocalizedData`.